jetpack/src/connection.rs

```diff
@@ -179,7 +166,7 @@ impl JetpackConnectionClient {
 
     pub async fn connect_user(
         &self,
-        wp_com_authentication: WpAuthentication,
+        auth_provider: Arc<WpAuthenticationProvider>,
```

Comment:
Unlike the authentication property in `WpApiClient` stored in the API client instance, this `wp_com_authentication` argument is discarded after use. I think we can leave it as `WpAuthentication`? Internally, we can wrap the `WpAuthentication` argument in a static auth `WpAuthenticationProvider`, which is then passed to  `WpApiClientDelegate`?

Reply:
I wasn't quite sure about the changes to the `wpcom` and `jetpack` package changes and didn't get a chance to fully investigate. I'll take a brief look at it tomorrow and will likely follow your advice.

Reply:
Addressed in cb53bcf433ff08453681df59bf219c4fd0224004.